envoy/tracing/trace_context.h

```diff
@@ -66,41 +70,33 @@ class TraceContext {
    * @param key The context key of string view type.
    * @return The optional context value of string_view type.
    */
-  virtual absl::optional<absl::string_view> getByKey(absl::string_view key) const PURE;
+  virtual absl::optional<absl::string_view> get(absl::string_view key) const PURE;
 
   /**
    * Set new tracing context key/value pair.
    *
    * @param key The context key of string view type.
    * @param val The context value of string view type.
    */
-  virtual void setByKey(absl::string_view key, absl::string_view val) PURE;
+  virtual void set(absl::string_view key, absl::string_view val) PURE;
 
   /**
-   * Set new tracing context key/value pair. The key MUST point to data that will live beyond
-   * the lifetime of any traceable stream that using the string.
+   * Removes the following key and its associated values from the tracing
+   * context.
    *
-   * @param key The context key of string view type.
-   * @param val The context value of string view type.
+   * @param key The key to remove if it exists.
    */
-  virtual void setByReferenceKey(absl::string_view key, absl::string_view val) PURE;
+  virtual void remove(absl::string_view key) PURE;
 
-  /**
-   * Set new tracing context key/value pair. Both key and val MUST point to data that will live
-   * beyond the lifetime of any traceable stream that using the string.
-   *
-   * @param key The context key of string view type.
-   * @param val The context value of string view type.
-   */
-  virtual void setByReference(absl::string_view key, absl::string_view val) PURE;
```

Comment:
reference semantics is only supported by HTTP and we should use it by the `TraceContextHandler` now.